samza-kv/src/main/java/org/apache/samza/storage/kv/BaseLocalStoreBackedTableDescriptor.java

```diff
@@ -57,9 +70,57 @@ public D withSideInputsProcessor(SideInputsProcessor sideInputsProcessor) {
     return (D) this;
   }
 
+  /**
+   * Disable changelog for this table, by default changelog is enabled.
+   * Refer to <code>stores.store-name.changelog</code> in Samza configuration guide
+   *
+   * @return this table descriptor instance
+   */
+  public D withChangelogDisabled() {
+    this.enableChangelog = false;
+    return (D) this;
+  }
+
+  /**
+   * Specify the changelog stream name, by default changelog stream name is automatically
+   * generated in format [job-name]-[job-id]-table-[table-id]. For both auto-generated
+   * and user specified changelog stream, invalid characters will be automatically
+   * replaced with '-' to confirm with pattern { @literal [a-zA-Z0-9_-]+ }
+   *
+   * Refer to <code>stores.store-name.changelog</code> in Samza configuration guide
+   *
+   * @param changelogStream changelog stream name
+   * @return this table descriptor instance
+   */
+  public D withChangelogStream(String changelogStream) {
```

Comment:
Do we have a use-case where it would be useful to override the changelogStream? 
Also, can we not rewrite the changelog stream names here but throw when the names don't conform to the pattern `@literal [a-zA-Z0-9_-]+`